scvi/dataset/_anndata.py

```diff
@@ -0,0 +1,337 @@
+        column_key = alternate_column_key
+    return column_key
+
+
+def _setup_protein_expression(
+    adata, protein_expression_obsm_key, protein_names_uns_key, batch_key
+):
+    assert (
+        protein_expression_obsm_key in adata.obsm.keys()
+    ), "{} is not a valid key in adata.obsm".format(protein_expression_obsm_key)
+
+    logger.info(
+        "Using protein expression from adata.obsm['{}']".format(
+            protein_expression_obsm_key
+        )
+    )
+    pro_exp = adata.obsm[protein_expression_obsm_key]
+    if _check_nonnegative_integers(pro_exp) is False:
+        warnings.warn(
+            "adata.obsm[{}] does not contain unnormalized count data. Are you sure this is what you want?".format(
+                protein_expression_obsm_key
+            )
+        )
+    # setup protein names
+    if protein_names_uns_key is None and isinstance(
+        adata.obsm[protein_expression_obsm_key], pd.DataFrame
+    ):
+        logger.info(
+            "Using protein names from columns of adata.obsm['{}']".format(
+                protein_expression_obsm_key
+            )
+        )
+        protein_names = list(adata.obsm[protein_expression_obsm_key].columns)
+    elif protein_names_uns_key is not None:
+        logger.info(
+            "Using protein names from adata.uns['{}']".format(protein_names_uns_key)
+        )
+        protein_names = adata.uns[protein_names_uns_key]
+    else:
+        logger.info("Generating sequential protein names")
+        protein_names = np.arange(adata.obsm[protein_expression_obsm_key].shape[1])
+
+    adata.uns["scvi_protein_names"] = protein_names
+
+    # batch mask totalVI
+    batch_mask = _get_batch_mask_protein_data(
+        adata, protein_expression_obsm_key, batch_key
+    )
+
+    # check if it's actually needed
+    if np.sum([~b for b in batch_mask]) > 0:
+        logger.info("Found batches with missing protein expression")
+        adata.uns["totalvi_batch_mask"] = batch_mask
+    return protein_expression_obsm_key
+
+
+def _setup_X(adata, X_layers_key):
+    # checking layers
+    if X_layers_key is not None:
+        assert (
+            X_layers_key in adata.layers.keys()
+        ), "{} is not a valid key in adata.layers".format(X_layers_key)
+        logger.info('Using data from adata.layers["{}"]'.format(X_layers_key))
+        X_loc = "layers"
+        X_key = X_layers_key
+        X = adata.layers[X_key]
+    else:
+        logger.info("Using data from adata.X")
+        X_loc = "_X"
+        X_key = None
+        X = adata._X
+
+    if _check_nonnegative_integers(X) is False:
+        logger_data_loc = (
+            "adata.X"
+            if X_layers_key is None
+            else "adata.layers[{}]".format(X_layers_key)
+        )
+        warnings.warn(
+            "{} does not contain unnormalized count data. Are you sure this is what you want?".format(
+                logger_data_loc
+            )
+        )
+    if adata.shape[0] < adata.shape[1] < 1:
```

Comment:
I'm not sure this condition is ever True, but I may be wrong. Is this a typo?

Reply:
ah good catch, yea that's a typo